students/BrandonHenson/Lesson9/Mailroom_5.py

```diff
@@ -0,0 +1,151 @@
+# Brandon Henson
+# 5/20/18
+# Lesson 09 mailroom_5.py
+# Fixing issues:Refactoring
+# !/usr/bin/env python3
+
+
+class Donor():
+    def __init__(self, name=None):
+        self._name = name
+        self._donations = []
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, val):
+        self._name = val
+
+    @property
+    def donations(self):
+        return self._donations
+
+    @donations.setter
+    def donations(self, val):
+        self._donations = val
+
+    @property
+    def total(self):
+        return sum(self.donations)
+
+
+class DonorAll(dict):
+    def __init__(self, donors):
+        self._donors = donors
+
+    @property
+    def donors(self):
+        return self._donors
+
+    @donors.setter
+    def donors(self, val):
+        self._donors = val
+
+    def add_donor(self, donor):
+        self.donors[donor.name] = donor.donations
+
+    def donate(self, name, donation):
+        if name not in self.donors.keys():
+            new_donor = Donor(name)
+            self.add_donor(new_donor)
+        else:
+            self.donors[name].append(int(donation))
+
+    def createfile(self, name):
+        for key, values in donor_history.items():
+            filename = str(key)+'.txt'
+            with open(filename, 'w') as fileobj:
+                total = sum(values)
+                numgifts = len(values)
+                fileobj.write("Dear {}\nThank you for your {} generous donations \
+totaling ${}\nThe money will be put to good use.\n\n\
+            Sincerely, \n                -The Team".format(key, numgifts, total))
+            fileobj.close()
+
+    def menu_2(self):
+        try:
+            print(f'{"Donor Name":18s} {"|  Total Given":12s} \
+    {"|  Num Gifts  |":14s} {"Average Gift":18s}')
+            print(f'{"-"*62}')
+            [print(f'{name:18s} ${sum(self.donors[name]):10.2f} {len(self.donors[name]):8d}        \
```

Comment:
it is preferred to use use regular for loop when you are discarding the return value.